druntime/src/core/internal/atomic.d

```diff
@@ -504,7 +509,7 @@ version (DigitalMars)
     T atomicFetchXor(MemoryOrder order = MemoryOrder.seq, T)(T* dest, T value) pure nothrow @nogc @trusted
     if (is(T : ulong))
     {
-        T set, get = atomicLoad!(order)(dest);
+        T set, get = atomicLoad!(MemoryOrder.raw)(dest);
```

Comment:
the order arg being passed down was something added somewhat recently in upstream ldc i think.... was this an intentional change or an accident?

Reply:
No, it is intentional and also a direct copy of the one you see in atomicOp. I don't know why the author of stdatomic added the ability to change the memory order of the initial load. I far as I can see the atomic load can be relaxed during the first load as the sequential order isn't that important there as long as it doesn't trespass the compareAndExchange. Specifically for x86, that ISA is an over achiever when it comes to sequential consistency and the ISA is known is as strongly ordered and it should be safe for x86. In fact for x86 it would be enough with just a mov which atomicLoad returns when it is only a MemoryOrder.raw load. atomicFetchXor is only for DMD which x86 only. On LDC it just uses the intrinsics instead for the same function.

When it comes to other architectures I'm a bit unsure but in that regard, it is the raw load in atomicOp that is wrong in that case.

This is really the same question as in this PR https://github.com/opendlang/opend/pull/33.

I would be good to have some input regarding this when it comes to the load on other ISAs than x86.

Reply:
Let's use C++ instead because of maturity.

https://godbolt.org/z/9MMP5KE5G

Here we can see regardless of memory order it produces the exact same code regardless of memory model, because of the nature of x86.